### PR TITLE
Disable LTPA cookie for SPNEGO authentication

### PR DESCRIPTION
**Pull request by utle**
Disable LTPA cookie for SPNEGO authentication

### CODE REVIEW

**Comment by utle** [comment]
#build

**Comment by andrea-pichler** [comment]
No message review requested from L2

File: dev/com.ibm.ws.security.spnego/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -58,3 +58,7 @@ includeClientGSSCredentialInSubject.desc=Specifies whether the client delegation
 authFilterRef=Authentication Filter Reference
 authFilterRef$Ref=Authentication filter reference
 authFilterRef.desc=Specifies the authentication filter reference. 
+
+disableLtpaCookie=Disable LTPA Cookie
+disableLtpaCookie.desc=Do not create an LTPA cookie during processing of the SPNEGO token. The LTPA Cookie does not add to the HTTP response.
```
Comment (dmuelle) [inline dev/com.ibm.ws.security.spnego/resources/OSGI-INF/l10n/metatype.properties:63]: `The LTPA Cookie does not add to the HTTP response.`

does this mean that the LTPA is not included in the response? In the current sentence, it's not clear what is not being added. Is it correct to say

`No LTPA cookie is added to the HTTP response.`

**Comment by utle** [comment]
#build

**Comment by utle** [comment]
#build

**Comment by utle** [comment]
#build

**Comment by utle** [comment]
#build

**Comment by utle** [comment]
#build

File: dev/com.ibm.websphere.security/src/com/ibm/ws/security/authentication/utility/SubjectHelper.java [hunk 2]
```diff
@@ -37,6 +37,12 @@
 public class SubjectHelper {
     private static final TraceComponent tc = Tr.register(SubjectHelper.class);
 
+    /**
```
Comment (teddyjtorres) [inline dev/com.ibm.websphere.security/src/com/ibm/ws/security/authentication/utility/SubjectHelper.java:40]: This comment seems to be extra and it appears that it was intended for the AuthenticationConstants.java class, which has this new comment already.

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/SSOCookieHelperImpl.java [hunk 3]
```diff
@@ -544,6 +547,27 @@ public void addSSOCookiesToResponse(Subject subject, HttpServletRequest req, Htt
             return;
         }
 
+        SubjectHelper subjectHelper = new SubjectHelper();
+
+        if (subjectHelper.getHashtableFromSubject(subject, disableSsoLtpaCookie) == null) {
```
Comment (teddyjtorres) [inline dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/SSOCookieHelperImpl.java:552]: There may be a defect in this path if someone adds AuthenticationConstants.INTERNAL_DISABLE_SSO_LTPA_COOKIE with a value of false to the hashtable. The cookie should still be created in that case.

Comment (utle) [reply dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/SSOCookieHelperImpl.java:552]: It's an internal property and default is false. I check for true any way

**Comment by utle** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.spnego_fat,com.ibm.websphere.security_fat,com.ibm.ws.collective.security_fat,com.ibm.ws.collective.security_zfat,com.ibm.ws.ejbcontainer.security.3.2_fat,com.ibm.ws.ejbcontainer.security.jacc_fat_java7,com.ibm.ws.ejbcontainer.security_fat,com.ibm.ws.ejbcontainer.security_fat.features,com.ibm.ws.security.audit_fat.config,com.ibm.ws.security.audit_fat.example,com.ibm.ws.security.audit_fat.features,com.ibm.ws.security.auth.data_fat,com.ibm.ws.security.authentication.builtin_fat,com.ibm.ws.security.authentication.tai_fat,com.ibm.ws.security.authorization.saf_zfat,com.ibm.ws.security.client_fat_java7,com.ibm.ws.security.context_fat,com.ibm.ws.security.credentials.saf_zfat,com.ibm.ws.security.csiv2_fat,com.ibm.ws.security.java2sec_fat,com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.oauth_fat,com.ibm.ws.security.openidconnect.client_fat,com.ibm.ws.security.openidconnect.client_fat.config,com.ibm.ws.security.openidconnect.client_fat.e2e,com.ibm.ws.security.openidconnect.client_fat.jaxrs,com.ibm.ws.security.openidconnect.client_fat.saml,com.ibm.ws.security.openidconnect.server_fat,com.ibm.ws.security.openidconnect.server_fat.config,com.ibm.ws.security.openidconnect.server_fat.endpoint.authorize,com.ibm.ws.security.openidconnect.server_fat.endpoint.clientregistration,com.ibm.ws.security.openidconnect.server_fat.endpoint.coveragemap,com.ibm.ws.security.openidconnect.server_fat.endpoint.discovery,com.ibm.ws.security.openidconnect.server_fat.endpoint.end_session,com.ibm.ws.security.openidconnect.server_fat.endpoint.introspect,com.ibm.ws.security.openidconnect.server_fat.endpoint.token,com.ibm.ws.security.openidconnect.server_fat.endpoint.userinfo,com.ibm.ws.security.openidconnect.server_fat.feature,com.ibm.ws.security.openidconnect.server_fat.granttype.jwt,com.ibm.ws.security.openidconnect.server_fat.jaxrs,com.ibm.ws.security.openidconnect.server_fat.jaxrs.config,com.ibm.ws.security.openidconnect.server_fat.oauth,com.ibm.ws.security.openidconnect.server_fat.oidc,com.ibm.ws.security.openidconnect.server_fat.saml,com.ibm.ws.security.quickstart_fat,com.ibm.ws.security.registry.saf_zfat,com.ibm.ws.security.registry_fat,com.ibm.ws.security.saml.sso_fat.config,com.ibm.ws.security.saml.sso_fat.config.mapToUserRegistry,com.ibm.ws.security.saml.sso_fat.endpoint.samlmetadata,com.ibm.ws.security.saml.sso_fat.jaxrs,com.ibm.ws.security.saml.sso_fat.jaxrs.config,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.spnego_fat,com.ibm.ws.security.thread.identity_fat,com.ibm.ws.security.thread.zos.context_zfat,com.ibm.ws.security.thread.zos_zfat,com.ibm.ws.security.token.ltpa_fat,com.ibm.ws.security.utility_fat,com.ibm.ws.security.wim.scim_fat,com.ibm.ws.security.wim.scim_zfat,com.ibm.ws.security_fat,com.ibm.ws.security_fat.features,com.ibm.ws.webcontainer.security.feature_fat,com.ibm.ws.webcontainer.security.jacc.1.5_fat,com.ibm.ws.webcontainer.security_fat,com.ibm.ws.webcontainer.security_fat.features,com.ibm.ws.webcontainer.security_fat.passwordutil,com.ibm.ws.security.openidconnect.client_fat.spnego,com.ibm.ws.security.openidconnect.server_fat.spnego

**Comment by utle** [comment]
My code change should not cause three tests failures in PB.

**Comment by utle** [comment]
#run-libby-bot

**Comment by utle** [comment]
#build

**Pull request merged**